src/python/turicreate/toolkits/object_detector/_tf_image_augmenter.py

```diff
@@ -22,38 +22,267 @@
+                        lambda: array_ops.reverse(uts[1], [0]),
+                        lambda: uts[1])
+    
+    flip_values_2 = math_ops.less(uts[2][1], uts[2][0])
+    uts[2] = control_flow_ops.cond(flip_values_2,
+                        lambda: array_ops.reverse(uts[2], [0]),
+                        lambda: uts[2])
+    
+    flip_values_3 = math_ops.less(uts[3][1], uts[3][0])
+    uts[3] = control_flow_ops.cond(flip_values_3,
+                        lambda: array_ops.reverse(uts[3], [0]),
+                        lambda: uts[3])
+
+    restack_points = tf.stack(uts)[:, :1]
+    reshaped_points = tf.reshape(restack_points, [ -1, 4])
+    
+    return tf.squeeze(reshaped_points)
+
+def _get_intersection_point(bounding_box1, bounding_box2):
+    intersection_cond = _is_intersection(bounding_box1, bounding_box2)
+    return control_flow_ops.cond(intersection_cond,
+                                lambda: _get_middle_points(bounding_box1, bounding_box2),
+                                lambda: np.array([0.0, 0.0, 0.0, 0.0], dtype=np.float32))
+def _get_area(bounds):
+    points = tf.unstack(bounds)
+    return ((points[2] - points[0]) * (points[3] - points[1]))
+
+def _crop_augmenter_loop_conditional(num_attempts, max_attempts, perform_cropped_annotations):
+    return math_ops.logical_and(math_ops.less_equal(num_attempts, max_attempts), math_ops.logical_not(perform_cropped_annotations))
+
+def _crop_augmenter_loop_perform_crop(image_box, keep_annotations, annotations, num_attempts, max_attempts, perform_cropped_annotations, aspect_ratio_var, min_aspect_ratio, max_aspect_ratio, min_height_var, max_height_var, image_height, image_width, min_area_fraction, max_area_fraction):
+    cropped_height = random_ops.random_uniform([], min_height_var, max_height_var)
+    cropped_width = cropped_height * aspect_ratio_var
+    
+    x_offset = random_ops.random_uniform([], 0.0, (image_width - cropped_width))
+    y_offset = random_ops.random_uniform([], 0.0, (image_height - cropped_height))
+        
+    unstacked_image_box = tf.unstack(image_box)
+    
+    unstacked_image_box[0] = tf.to_float(tf.to_int32(y_offset))
+    unstacked_image_box[1] = tf.to_float(tf.to_int32(x_offset))
+    unstacked_image_box[2] = tf.to_float(tf.to_int32(cropped_height))
+    unstacked_image_box[3] = tf.to_float(tf.to_int32(cropped_width))
+
+    image_box = tf.stack(unstacked_image_box)
+    
+    intersection_image_box = tf.unstack(image_box)
+    
+    intersection_image_box[0] = tf.to_float(tf.to_int32(y_offset))
+    intersection_image_box[1] = tf.to_float(tf.to_int32(x_offset))
+    intersection_image_box[2] = tf.to_float(tf.to_int32(cropped_height + y_offset))
+    intersection_image_box[3] = tf.to_float(tf.to_int32(cropped_width + x_offset))
+
+    intersection_bounding_box = tf.stack(intersection_image_box)
+    
+    identifier, box, confidence = decompose_annotations(annotations, tf.to_float(image_height), tf.to_float(image_width))
```

Comment:
Most of the time, annotation (singular) is passed into `decompose_annotations`, but here it's plural. How does it work and is `annotations` here really multiple boxes?